### DISCUSSION TITLE
How can I get initial zoom value after graph has been drawn. Or how to draw zoomable decorations

### DISCUSSION THREAD

**Discussion by MegaAndroidDeveloper666** [patrykandpatrick/vico#612] [Questions]
Hello! 
I'm using latest alpha version 2.0.0-alpha11
Initial issue is "highlighting some parts of graph". To achieve that I used decorations. Works great, but graph could be zoomable, so I need to zoom and offset decoraions also. 
So two questions:
1) Is there a way to kind of automatically draw zoomable and scrollable decorations (anchored to graph window bounds and graph itself)?
2) If no, how do I get initial graph zoom value? Or how to catch the moment of graph initially drawn? So I could calculate decorations' offset and size

**Comment by patrickmichalik** [comment]
Hello! `Decoration`s have to handle scroll and zoom themselves. (We’ll be looking into adding a helper API.) The current scroll value and zoom factor can be read via the `horizontalScroll` and `zoom` fields of `ChartDrawContext`, an instance of which is passed to both `Decoration` functions.
